### PR TITLE
There are now walls to collide with

### PR DESCRIPTION
**Pull request by FredTheDino**
This is probably all the collision code we will need for the rest of the jam. It's great because it has no tunneling (for reasonable speeds). It's also mathematically exact - so our circles really are circles. No shitty "segmented wanna be circles" here!

I managed to get some tunnelning with the puckspeed ~ 100, but I think if we keep below 75 we'll be fine!

It's great if you try to find problems with it - it should be robust but it might have literal edge-cases.

### CODE REVIEW

File: game.sy [hunk 1]
```diff
@@ -238,6 +270,95 @@ update :: fn delta: float -> void {
...
+            b: collision.Wall {
+                pos: (0.0, 0.0),
+                normal: (0.0, 0.0),
+            },
+        }
+
+        // Find next collision
+        a_i := 0
+        // TODO(ed): Write this with for_each, when the upvalue bug is fixed.
+        loop a_i < len(bodies) {
```
Comment (FredTheDino) [inline game.sy:294]: This is a loop because of a language bug - I didn't manage to fix this one while I was at it. :(

File: game.sy [hunk 2]
```diff
@@ -238,6 +270,95 @@ update :: fn delta: float -> void {
...
+            }
+            if :collision.Body is next_col.b {
+                b :!collision.CircleBody = next_col.b
+                b.pos -= n * 0.01
+                b.vel = energy_kept * reflect' b.vel, n
+            }
+        }
+        t -= step
+    }
+    if iter == 0 { <!> }
```
Comment (FredTheDino) [inline game.sy:360]: This shouldn't be in the final game - but I put it here so we see when we run out of collisions.

File: game.sy [hunk 3]
```diff
@@ -238,6 +270,95 @@ update :: fn delta: float -> void {
         p.try_grab' puck
     }
 
+    iter := 1000
```
Comment (FredTheDino) [inline game.sy:273]: If we hit this cap - we crash - that's deliberate. I don't think we can allocate much more than 1000 collision steps without dropping frames.

File: game.sy [hunk 4]
```diff
@@ -203,17 +213,39 @@ init :: fn {
...
+    }
+
+    push' corners, collision.Circle {
+        rad: d,
+        pos: (0.0, 1.0 - d),
+    }
+    push' corners, collision.Circle {
+        rad: d,
+        pos: (0.0, -1.0 + d),
+    }
```
Comment (FredTheDino) [inline game.sy:248]: This is placeholder stuff. It creates and arena with a rounded top and bottom and straight walls on the sides.

File: game.sy [hunk 5]
```diff
@@ -64,12 +75,11 @@ player_new :: fn pos: (float, float) -> Player {
         },
         process_input: fn input: Input, delta: float -> void {
 
-            shoot_speed :: self.vel + 2.0
+            shoot_speed :: self.vel + 2.0 + 50.0
```
Comment (EMattfolk) [inline game.sy]: Maybe lower this?
```suggestion
            shoot_speed :: self.vel + 2.0
```

**Review by EMattfolk — APPROVED** [review]

**Pull request merged**